hedera-node/src/main/java/com/hedera/services/tokens/HederaTokenStore.java

```diff
@@ -259,6 +231,13 @@ public ResponseCodeEnum wipe(AccountID aId, TokenID tId, boolean skipKeyCheck) {
 			}
 			var account = ledger.getTokenRef(aId);
 			var balance = account.getTokenBalance(tId);
+
+			if(wipingAmount < 0 || wipingAmount > balance) {
+				return INVALID_WIPING_AMOUNT;
+			}
+
+			balance -= wipingAmount;
+
 			var outcome = account.wipeTokenRelationship(tId);
 			if (outcome == OK) {
 				ledger.markForMerge(aId);
```

Comment:
As per the specification -> The wiped tokens must be burned and not transferred to the treasury.

So in my view:
1) The account holder must get his balance changed
2) The balance difference must be burned => the Token's float must be adjusted